src/modules/commander/companion_process_status.cpp

```diff
@@ -0,0 +1,155 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2013-2016 PX4 Development Team. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/**
+ * @file companion_process_status.cpp
+ * Monitor health status of companion processes
+ */
+
+#include <drivers/drv_hrt.h>
+#include "companion_process_status.h"
+using uORB::Subscription;
+
+static orb_advert_t *mavlink_log_pub;
+
+
+Companion_Process_Status::Companion_Process_Status(){
+
+	//initialize time
+	_time_zero = hrt_absolute_time();
+	_time_message = hrt_absolute_time();
+}
+
+void Companion_Process_Status::poll_subscriptions(){
+
+    if (_companion_process_status_sub.update()) {
+		_companion_process_status = _companion_process_status_sub.get();
+		_new_status_received = true;
+	}
+}
+
+void Companion_Process_Status::determine_required_processes(int32_t use_obs_avoid){
+
+	//determine required processes from parameters
+	_avoidance_required = use_obs_avoid;
+
+}
+
+void Companion_Process_Status::determine_action(){
+
+	bool reported_before = false;
+	companion_process_status_s last_status;
+	_process_type = UNDEFINED;
+
```

Comment:
Shouldn't we add a check for the `ONBOARD_CONTROLLER` type of HEARBEAT's are getting received as well? That would be redundant, but would guarantee we have a check for both link and process status.

Reply:
I found some heartbeats of MAV_TYPE = 18 (onboard)
and COMPONENT_ID = 240 (udp_bridge)
Do you mean those? will they always use component 240 even if you go over serial? Or is 241 also valid?

Reply:
yes they will always use those

Reply:
240 usually means you have QGC plugged on the stream through the MAVROS, while 18 is sent from MAVROS `sys_status` plugin.

Reply:
But the check for those should be done on the commander state machine. Check https://github.com/PX4/Firmware/pull/7985 for a reference.

Reply:
makes sense. But it seems like this s already implemented under the PR you referenced. Would it not make more sense to clean up and merge that one instead of reimplementing all of that under this PR?

Reply:
Well that PR is a little outdated and needs some cleanup. We can bring it again on a new PR. Let's stay this one as is and bring the features of the above later.